lib/Parse/ParseDecl.cpp

```diff
@@ -186,7 +186,9 @@ void PersistentParserState::parseMembers(IterableDeclContext *IDC) {
   SourceFile &SF = *IDC->getDecl()->getDeclContext()->getParentSourceFile();
   assert(!SF.hasInterfaceHash() &&
     "Cannot delay parsing if we care about the interface hash.");
+  assert(SF.Kind != SourceFileKind::SIL && "cannot delay parsing SIL");
```

Comment:
Why not?

Reply:
We need to have a way to keep `SILParserState` around till delayed parsing, which requires non-trivial work. I don't think there're fundamental blockers there, just we need to keep track more stuff.

Reply:
Ok, fair enough. Can you add a comment somewhere to that effect? We don’t care about SIL parsing performance much anyway, I was just curious.